aten/src/ATen/functorch/BatchRulesHelper.cpp

```diff
@@ -182,18 +171,6 @@ std::tuple<Tensor, Tensor> _binary_pointwise_helper(
   auto tensor_ = moveBatchDimToFront(tensor, tensor_batch_dim);
   auto other_ = moveBatchDimToFront(other, other_batch_dim);
 
-  // In the (0D, ND) case, type promotion semantics are different :/
-  if (do_type_promotion) {
```

Comment:
Do you know what was this code doing before? Was it just plain wrong? It would be good to test it on regular functorch (no torch compile)

Reply:
AFAIK 0D tensors have different type promotion rules so when applying vmap we can't rely on the underlying operator to do type promotion since it will see a 1D tensor. So I think this code does need to stay.